cmd/manager/main.go

```diff
@@ -17,72 +17,20 @@ package main
 
 import (
 	"flag"
-	"os"
+	"log"
 
-	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
-	_ "k8s.io/client-go/plugin/pkg/client/auth"
-	"knative.dev/serving-operator/pkg/apis"
-	"knative.dev/serving-operator/pkg/reconciler"
-
-	"github.com/spf13/pflag"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+	"k8s.io/client-go/tools/clientcmd"
+	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/signals"
+	"knative.dev/serving-operator/pkg/reconciler/knativeserving"
 )
 
-var log = logf.Log.WithName("cmd")
-
 func main() {
-	// Add flags registered by imported packages (e.g. glog and
-	// controller-runtime)
-	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
-
-	pflag.Parse()
-
-	// Use a zap logr.Logger implementation. If none of the zap
-	// flags are configured (or if the zap flag set is not being
-	// used), this defaults to a production zap logger.
-	//
-	// The logger instantiated here can be changed to any logger
-	// implementing the logr.Logger interface. This logger will
-	// be propagated through the whole operator, generating
-	// uniform and structured logs.
-	logf.SetLogger(logf.ZapLogger(true))
+	flag.Parse()
 
-	// Get a config to talk to the apiserver
-	cfg, err := config.GetConfig()
+	cfg, err := clientcmd.BuildConfigFromFlags(*knativeserving.MasterURL, *knativeserving.Kubeconfig)
 	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	// Create a new Cmd to provide shared dependencies and start components
-	mgr, err := manager.New(cfg, manager.Options{})
-	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	log.Info("Registering Components.")
-
-	// Setup Scheme for all resources
-	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	// Setup all Controllers
-	if err := reconciler.AddToManager(mgr, cfg); err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	log.Info("Starting the Cmd.")
-
-	// Start the Cmd
-	if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
-		log.Error(err, "Manager exited non-zero")
-		os.Exit(1)
+		log.Fatal("Error building kubeconfig", err)
 	}
+	sharedmain.MainWithConfig(signals.NewContext(), "controller", cfg, knativeserving.NewController)
```

Comment:
I dont think we can should "controller" as the component name as this is already taken (although in another namespace). I also worry that the name is not really specific enough for the default namespace (pretty much anything can be called controller in the deafault ns). Can we use something like `servingoperator`

Reply:
@greghaynes Done.